data_linter/validators/pandas_validator.py

```diff
@@ -79,22 +80,159 @@ def read_data(self) -> pd.DataFrame:  # KARIK TODO
                 df = pa_read_csv_to_pandas(
                     input_file=f,
                     schema=None,  # Needs actual schema
-                    expect_full_schema=False
+                    expect_full_schema=False,
                 )
             elif "json" in self.metadata.data_format:
                 df = pa_read_json_to_pandas(
                     input_file=f,
                     schema=None,  # Needs actual schema
-                    expect_full_schema=False
+                    expect_full_schema=False,
                 )
             elif "parquet" in self.metadata.data_format:
-                df = arrow_to_pandas(
-                    pq.read_table(f)
-                )
+                df = arrow_to_pandas(pq.read_table(f))
             else:
-                raise ValueError(f"Unknown data_format in metadata: {self.metadata.data_format}.")
+                raise ValueError(
+                    f"Unknown data_format in metadata: {self.metadata.data_format}."
+                )
         return df
 
-
     def validate_df(self, df):  # STEPHEN TODO
+
+        meta_cols = [col for col in self.metadata["columns"] if col["name"] in df]
+
+        for i, (_, col) in enumerate(df.iteritems()):
+            self.validate_col(col, meta_cols[i])
```

Comment:
for m in meta_cols:
    self.validate_col(col, meta_cols[i])
```suggestion
        for m in meta_cols:
            self.validate_col(df[m["name"]], m)
```